refactor user role classmethods to utils package

snopoke opened the pull request:
## Summary
Reduce the scope of the `UserRole` class by moving most of the class level methods to standalone functions.

This is part of ongoing work for moving `UserRole` to SQL.

Can review by commit.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
Some added, but mostly just moving code

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

snopoke committed:
new 'UserRole.create' which is simpler than 'get_or_create_with_permissions'

snopoke committed:
remove permissions where they aren't needed

snopoke committed:
make name arg required

snopoke committed:
extract function with tests

snopoke committed:
move get_read_only_role_for_domain to utils

snopoke committed:
move get_custom_roles_by_domain

snopoke committed:
move [archive/unarchive]_roles_for_domain

snopoke committed:
move reset_initial_roles_for_domain

snopoke committed:
move init_domain_with_presets

snopoke committed:
get_all_role_names_for_domain

orangejenny commented on corehq/apps/accounting/subscription_changes.py:
```diff
@@ -243,7 +243,7 @@ def response_role_based_access(domain, new_plan_version):
         # temporarily disable this part of the downgrade until we
         # have a better user experience for notifying the downgraded user
-        # read_only_role = UserRole.get_read_only_role_by_domain(self.domain.name)
+        # read_only_role = get_read_only_role_for_domain(self.domain.name)
```
How about just deleting this? It's been commented out since 2014: https://github.com/dimagi/commcare-hq/commit/bf14e6cce27b49e203796e260b58adf410b4eaed

snopoke committed:
remove get_read_only_role_for_domain and related code

This has been commented out for some years so unlikely to be needed
https://github.com/dimagi/commcare-hq/commit/bf14e6cce27b49e203796e260b58adf410b4eaed

snopoke committed:
replace missed usage

snopoke committed:
create test role in correct domain
